### PR TITLE
be: replace interpreter by interpreter2

### PR DESCRIPTION
**Pull request by michaellilltokiwa**

### CODE REVIEW

File: src/dev/flang/be/interpreter/Interpreter.java [hunk 1]
```diff
@@ -24,230 +24,100 @@
...
 {
+  public static AbstractInterpreter<Value, Object> _ai;
+  // NYI: HACK:
+  public static Interpreter instance;
 
+  public final FUIR _fuir;
+  public final FuzionOptions _options_;
+  private final Excecutor _processor;
 
-  /*-----------------------------  statics  -----------------------------*/
```
Comment (fridis) [inline src/dev/flang/be/interpreter/Interpreter.java:81]: I think it helps to keep the general structure of classes, in particular of large classes. So please use these separators.

**Review by fridis — CHANGES_REQUESTED** [review]
This is a big step ahead!

May comments are mostly about code structure and comments that in parts is IMHO worse than in the old interpreter, but probably easy to fix. 

Also, please add an issue to add support for stack traces of a quality similar to the old interpreter. This is a feature I will miss.

File: tests/contracts_negative/SquareRoot.fz.expected_err [hunk 2]
```diff
@@ -1,11 +1,4 @@
 
---CURDIR--/SquareRoot.fz:30:9: error 1: Precondition does not hold
-      a ≥ 0
---------^^^
-For call to SquareRoot.sqrt
-Call stack:
-SquareRoot: --CURDIR--/SquareRoot.fz:45:21:
-  say "sqrt(-16) = {sqrt(-16)}" // 1. should flag an error, precondition failed
---------------------^^^^
+error 1: CONTRACT FAILED: pre-condition on call to 'SquareRoot.sqrt'
```
[the remark this answers is not in the document]
Comment (michaellilltokiwa) [reply tests/contracts_negative/SquareRoot.fz.expected_err:2]: Yes, I agree. Will add an issue for this.

**Review by fridis — APPROVED** [review]
Looks good!
Please keep an eye on the jenkins flang.dev results and fix any new issues that will arise (like missing stack traces in err output etc.). Keep an eye on the change of total time spent in the interpreter, might make sense to use `-XjavaProf` to understand what is the most performance critical part.

**Pull request merged**